### PR TITLE
Engagement cohorts UI

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
- UI necessary for #4510 
- opening draft for review purposes
- behind feature flag `engagement-cohort`
*If this affects the frontend, include screenshots.*  
Screen Shot 2021-06-02 at 11 57 19 AM
## Checklist

- [ ] More styling
- [ ] Put behind feature flag
- [ ] Add tags

### CODE REVIEW

**Comment by EDsCODE** [comment]
based on wireframes #4349 

Note: Opted to not change the property filter UI yet because the propertyfilter component we have currently was easily reuasble. As a follow-up can create a new component and make that switch

File: frontend/src/types.ts [hunk 1]
```diff
@@ -294,13 +294,22 @@ export interface PersonType {
...
     action_id?: number
+    event_id?: string
+    label?: string
+    count?: number
+    count_operator?: string
     properties?: Record<string, any>
+    matchType: MatchType
 }
 
+export type MatchType = 'entities' | 'properties'
```
Comment (paolodamico) [inline frontend/src/types.ts:308]: We also use these as constants, should we just declare them in a single place to simplify?

**Review by paolodamico — COMMENTED** [review]
Alright, took a long look at this and improving the full UI, and I think we're almost good to go. 

A few outstanding I need your help with:
1. Screen is frozen while cohort is recalculating which could take a long time. I think we should get rid of the toast and just show calculating in the matched users section. We should also allow the user to close the drawer.
2. Navigating to `/cohorts/new` doesn't open the drawer.
3. Attempting to create a cohort with the default config (i.e. just setting a name) leads to a 500 error, but we don't restrict anything on the frontend.
4. There seems to be some bug / performance issue rendering the matched users list as my screen freezes for like 7-10 seconds when opening a cohort (see recording), we might need to consider a virtualized list.
image

**Comment by paolodamico** [comment]
Alright master rebased and merged. Fixed issues that popped up from that. I'll now take a look at the outstanding issues to get this in.

**Comment by paolodamico** [comment]
Alright @EDsCODE ready for a thorough look before this becomes even larger. I think you're the only one with enough context to review.

FYI I added this state when the cohort is being calculated.


And this error message if you add an empty group.

**Comment by paolodamico** [comment]
Nudge @EDsCODE to avoid more merge conflicts

**Comment by paolodamico** [comment]
Fixed merge conflicts and re-tested non-feature-flagged behavior, looks like it's working as expected. Will merge to avoid another round of conflicts and we can start testing the new feature behind the feature flag.

**Pull request merged**